src/test/java/seedu/address/logic/commands/edit/EditInternshipCommandTest.java

```diff
@@ -1,2 +1,187 @@
+import static seedu.address.testutil.TypicalIndexes.INDEX_THIRD_ITEM;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.commands.results.CommandResult;
+import seedu.address.model.item.Internship;
+import seedu.address.testutil.InternshipBuilder;
+import seedu.address.testutil.ModelStub;
+import seedu.address.testutil.TypicalInternship;
+
+public class EditInternshipCommandTest {
+
+    private EditInternshipDescriptor editInternshipDescriptor = new EditInternshipDescriptor();
+
+    Internship sampleEditedInternship = TypicalInternship.GOOGLE;
+
+    @Test
+    public void constructor_nullInternship_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new EditInternshipCommand(null, null));
+    }
+
+    @Test
+    public void execute_invalidIndex_throwsCommandException() {
+        Internship validItem = new InternshipBuilder().build();
+        ModelStub modelStub = new EditInternshipCommandTest.ModelStubWithInternship(validItem);
+        Index invalidIndex = INDEX_THIRD_ITEM;
+        setEditInternshipDescriptor();
+        EditInternshipCommand editInternshipCommand = new EditInternshipCommand(invalidIndex, editInternshipDescriptor);
+        assertThrows(CommandException.class,
+                Messages.MESSAGE_INVALID_INDEX, () -> editInternshipCommand.execute(modelStub));
+    }
+
+    @Test
+    public void execute_validIndex_editSuccessful() throws CommandException {
+        ModelStubContainingInternshipEdited modelStub = new ModelStubContainingInternshipEdited(sampleEditedInternship);
+        setEditInternshipDescriptor();
+        Index validIndex = INDEX_FIRST_ITEM;
+
+        Internship toEdit = modelStub.getInternshipByIndex(validIndex);
+        EditInternshipCommand editInternshipCommand = new EditInternshipCommand(validIndex, editInternshipDescriptor);
+        assertEquals(Arrays.asList(toEdit), modelStub.internships);
+
+        CommandResult commandResult = editInternshipCommand.execute(modelStub);
+
+        assertEquals(String.format(EditInternshipCommand.MESSAGE_EDIT_INTERNSHIP_SUCCESS, sampleEditedInternship),
+                commandResult.getFeedbackToUser());
+    }
+
+    @Test
+    public void execute_duplicateInternship_throwsCommandException() throws CommandException {
+        Index validIndex = INDEX_FIRST_ITEM;
+        setEditInternshipDescriptor();
+        EditInternshipCommand editCommand = new EditInternshipCommand(validIndex, editInternshipDescriptor);
+        ModelStubContainingInternshipEdited modelStub = new ModelStubContainingInternshipEdited(
+                new InternshipBuilder().build());
+        modelStub.addInternship(sampleEditedInternship);
+
+//      I'm actually not sure how to test this.
```

Comment:
Check out `seedu.address.testutil.Assert.assertThrows`.

Also try to avoid leaving messages like this lest you forget to remove them later.

Reply:
Thanks! I tried assertThrows but it doesn't throw exception if I edit Internship 1's name into Internship 2's name which is supposed to cause DuplicateItemException. That's why I leave the assertThrows out so it can pass for now